Add documentation and a template for Great First Issues

mrjoro opened the pull request:
Adds documentation for creators of Great First Issues and a template to use when creating Great First Issues.

cc @adelinamart

cramforce commented on contributing/creating-great-first-issues.md:
```diff
@@ -0,0 +1,22 @@
+
+The AMP Project welcomes new contributors and we want to make it as easy as possible for them to contribute.  For many new contributors (who may not have open source/Git/AMP/etc. experience) it can be difficult to figure out how to get started.
+
+To help these new contributors get oriented we curate [Great First Issues](https://github.com/ampproject/amphtml/issues?q=is%3Aopen+is%3Aissue+label%3A%22Great+First+Issues%22).  A Great First Issue is a starter issue that a new contributor can use to get comfortable contributing to the AMP Project.
```
This is great. One thing I'm wondering:

Should we rename our current label to "Candidate for Great First Issue"? I think it would be helpful if engineers could add this label with low friction, and then we only use the real label for those issues that have been fully curated.

mrjoro replied:
Let's bring the conversation from @adelinamart into here so it's in context. :)

I'm fine with either approach (milestone vs. label); in any case I will add the details for that once we decide which to go with.  @adelinamart wdyt?

adelinamart replied:
My recommendation will be to use a milestone. 
Indeed both a label or a milestone are one click away and they are both very explicit, but with milestone you actually create a specific action queue for this purpose with a timeframe if needed. Also, there are so many labels that it might get confusing having 2 similar labels.

cramforce replied:
Yeah, it makes sense to use a label for the candidates and a milestone for the prepared issues.

mrjoro replied:
We spoke in person and decided on the approach Malte suggested.  I changed the milestone to be called Great First Issues, and will change the label to GFI Candidates once this PR goes in.

adelinamart commented on contributing/great-first-issues-template.md:
```diff
@@ -0,0 +1,64 @@
+<!--
+List knowledge the contributor should have before working on this issue.  You can mention that some of these are nice-to-have if you think the knowledge helps but that the issue is still relatively straightforward without that knowledge.
+-->
+### What you will need to know
```
I feel 'what you will ned to know' , 'background', 'motivation' and 'the bug' might have overlapping context. Personally I will feel overwhelmed to see so much text to go through and I will probably not read it :) . Maybe we can group some of these and keep it shorter and to the point.

mrjoro replied:
Take a look at an example from Hoodie (which inspired this):

https://github.com/hoodiehq/camp/issues/98

I think in general each section has an important reason for being there, but I wasn't as clear about that as I could have been.  I've updated the comments to make it more clear how they are distinct/why each one is important.

adelinamart commented:
LGTM. I left a short comment. @cramforce I was suggesting not use a different label, but a milestone instead for the ones that needs to be reviewed first, because it might get confusing to have 2 similar labels. Since the idea of having the issue reviewed for being a 'great first issue' is an action with a determined time, a milestone will be more appropriate.

cramforce commented:
No strong feelings, but I think a label is better, because it can be applied before we are sure on prioritization.

mrjoro commented:
I updated the docs to reflect the change to a milestone for the GFIs and a label for candidates.  I added a couple of files into this PR to update the links to point to the milestone.

cramforce commented on contributing/creating-great-first-issues.md:
```diff
@@ -0,0 +1,24 @@
+# Creating Great First Issues
```
Would be good to link to this file from somewhere.
